data-reconciliation-app/src/repositories/kv-repository.ts

```diff
@@ -0,0 +1,69 @@
+import * as ccfapp from "@microsoft/ccf-app";
+
+export interface IKeyValueRepository<T> {
+  set(key: string, value: T): T;
+  get(key: string): T;
+  has(key: string): boolean;
+  keys(): string[];
+  values(): T[];
+  get size(): number;
+}
+
+// generic key-value repository wrapping ccf TypedKvMap interaction
+export class KeyValueRepository<T> implements IKeyValueRepository<T> {
+  private kvStore: ccfapp.TypedKvMap<string, T>;
+
+  public constructor() {
+    this.kvStore = this.getDataMap();
+    console.log(this.kvStore);
+  }
+
+  // update key-value pair or create new record if key not exists
+  public set(key: string, value: T): T {
+    this.kvStore.set(key, value);
```

Comment:
The repository should handle any exceptions thrown from `kvStore` and return `ServiceResult`

Reply:
I think we don't expect many exceptions from kvstore, and if it happens, we will handle it on the API layer.
Beside that, returning the value directly from the repository will give simple and readable code, which will be suitable for the samples repository.

How does it sound?

Reply:
Handling exception at the place of happening is much more readable code and much more maintainable. When an exception bubbles up all the way to API layer, that is an invitation for bad SLA of the service.

I highly recommend handling it here for that reason :)

Reply:
I feel it makes the code more focused on checking the result of each call (to inspect the exceptions) instead of trusting the provided implementation. From my point of view, I think it is better to treat exceptions as exceptional and handle them on a certain layer, excluding business logic validation error from that.

I will implement it as you mention, and later we can have a deeper discussion.